keras_cv/models/segmentation/deeplab.py

```diff
@@ -82,110 +97,95 @@ def __init__(
-                        "stackwise_strides"
-                    ],
-                    stackwise_dilations=BACKBONE_CONFIG["ResNet50V2"][
-                        "stackwise_dilations"
-                    ],
-                    include_rescaling=include_rescaling,
-                    include_top=False,
-                    name="resnet50v2",
-                    weights=parse_weights(backbone_weights, False, "resnet50v2"),
-                    pooling=None,
-                    **kwargs,
+                if backbone_weights and input_shape[-1] == 1:
+                    raise ValueError(
+                        "The input shape is set up for greyscale images with one channel, but backbone weights are trained on colored images and cannot be loaded."
+                    )
+                backbone = get_resnet_backbone(
+                    backbone_weights, include_rescaling, input_shape, **kwargs
                 )
 
         else:
             # TODO(scottzhu): Might need to do more assertion about the model
+            # What else do we want to test for? Shapes? This feels like too little, but
+            # more assertions feel like they'd be limiting.
             if not isinstance(backbone, tf.keras.layers.Layer):
                 raise ValueError(
                     "Backbone need to be a `tf.keras.layers.Layer`, "
                     f"received {backbone}"
                 )
-        self.backbone = backbone
 
+        feature_map = backbone(x)
         if spatial_pyramid_pooling is None:
-            self.aspp = SpatialPyramidPooling(dilation_rates=[6, 12, 18])
-        else:
-            self.aspp = spatial_pyramid_pooling
+            spatial_pyramid_pooling = SpatialPyramidPooling(dilation_rates=[6, 12, 18])
+
+        output = spatial_pyramid_pooling(feature_map)
+        output = tf.keras.layers.UpSampling2D(
+            size=(height // feature_map.shape[1], width // feature_map.shape[2]),
+            interpolation="bilinear",
+        )(output)
 
-        self._segmentation_head_passed = segmentation_head
         if segmentation_head is None:
-            segmentation_head = tf.keras.Sequential(
-                [
-                    tf.keras.layers.Conv2D(
-                        filters=256,
-                        kernel_size=(1, 1),
-                        padding="same",
-                        use_bias=False,
-                    ),
-                    tf.keras.layers.BatchNormalization(),
-                    tf.keras.layers.Activation("relu"),
-                    tf.keras.layers.Dropout(0.2),
-                    tf.keras.layers.Conv2D(
-                        filters=classes,
-                        kernel_size=(1, 1),
-                        padding="same",
-                        use_bias=False,
-                        activation="softmax",
-                        # Force the dtype of the classification head to float32 to avoid the NAN loss
```

Comment:
This "forcing to float32" part needs to be kept.

Reply:
It's still there, but at a different place in the file